WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java

```diff
@@ -1,60 +1,110 @@
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onBackPressed() {
+        // perform the Nav back first so updated fragments can find their expected state there.
+        retrieveLoginNavFragment().goBack();
+
+        super.onBackPressed();
+    }
+
+    private LoginNavFragment retrieveLoginNavFragment() {
+        Fragment loginNavFragment = getSupportFragmentManager().findFragmentByTag(TAG_LOGIN_NAV_FRAGMENT);
+        if (loginNavFragment == null) {
+            return null;
+        } else {
+            return (LoginNavFragment) loginNavFragment;
+        }
+    }
+
+    private void addLoginNavFragment() {
+        LoginNavFragment loginNavFragment = retrieveLoginNavFragment();
+
+        if (loginNavFragment == null) {
+            loginNavFragment = new LoginNavFragment();
+
+            FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+            fragmentTransaction.add(loginNavFragment, TAG_LOGIN_NAV_FRAGMENT);
+            fragmentTransaction.commit();
+        }
+    }
+
     protected void addLoginPrologueFragment() {
         LogInOrSignUpFragment loginSignupFragment = new LogInOrSignUpFragment();
         FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
         fragmentTransaction.replace(R.id.fragment_container, loginSignupFragment, LogInOrSignUpFragment.TAG);
         fragmentTransaction.commit();
     }
 
-    @Override
-    public LoginEvents.LoginNavPrologue getLoginNavPrologue() {
-        return mLoginNavigationController.getLoginNavPrologue();
+    private void slideInFragment(Fragment fragment, String tag) {
+        slideInFragment(fragment, true, tag);
     }
 
-    @Override
-    public LoginEvents.LoginNavInputEmail getLoginNavInputEmail() {
-        return mLoginNavigationController.getLoginNavInputEmail();
+    private void slideInFragment(Fragment fragment, boolean shouldAddToBackStack, String tag) {
+        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.setCustomAnimations(R.anim.activity_slide_in_from_right, R.anim.activity_slide_out_to_left,
+                R.anim.activity_slide_in_from_left, R.anim.activity_slide_out_to_right);
+        fragmentTransaction.replace(R.id.fragment_container, fragment, tag);
+        if (shouldAddToBackStack) {
+            fragmentTransaction.addToBackStack(null);
+        }
+        fragmentTransaction.commitAllowingStateLoss();
     }
 
     @Override
-    public LoginEvents.LoginNavInputSiteAddress getLoginNavInputSiteAddress() {
-        return mLoginNavigationController.getLoginNavInputSiteAddress();
+    public LoginStateGetter getLoginStateGetter() {
+        return retrieveLoginNavFragment().getLoginStateGetter();
```

Comment:
Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/5848#discussion_r116673071.